thorlcr/activities/msort/thmsortslave.cpp

```diff
@@ -187,10 +187,7 @@ class MSortSlaveActivity : public CSlaveActivity
     }
     virtual void serializeStats(MemoryBuffer &mb) override
     {
-        {
-            CriticalBlock block(statsCs);
-            mergeStats(stats, sorter, spillStatistics);
-        }
+        collateStats([&](CRuntimeStatisticCollection &activeStats) { mergeStats(activeStats, sorter, spillStatistics); });
```

Comment:
possibly cleaner/simpler to have a 
gatherActiveStats() virtual

Reply:
agree on balance, have changed.